### PR TITLE
feat: implementation of the wish import, similar to paimon.moe

### PR DESCRIPTION
**Pull request by LudovicMalot**

### CODE REVIEW

File: src/utils/wishImport.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	for (const gachaType of gachaTypeList) {
+		const bannerHistory = [];
+		params.gacha_type = Number(gachaType.key);
+		let currentPage = 1;
+		let queryCount = 0;
+
+		while (true) {
+			params.page = currentPage;
+			try {
+				const response = (await axios.get(url, { params })).data as hoyoWishResponse;
```
Comment (maalni) [inline src/utils/wishImport.ts:42]: Dont assert Types.

```suggestion
				const response = (await axios.get<hoyoWishResponse>(url, { params })).data;
```

File: src/utils/wishImport.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+						break;
+					}
+					if (gachaList.length === 0) {
+						break;
+					}
+					currentPage++;
+					queryCount++;
+					if (queryCount % 5 === 0) {
+						await delay(1000);
+					}
```
Comment (maalni) [inline src/utils/wishImport.ts:66]: Why delay every 5th page? Wouldnt random delay between 100-1000ms be better to avoid detection?

**Review by maalni — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
+});
+// exemple request : http://localhost:3000/wishhistory?authkey=MYO%2BmxJzieyA8xPITwQSZCl%2BMFm1fMrf9zZ9e4IIKaPrjlGnTw4U2nMOH4YTKYhKXf2d4NEPh5Vap49rDhGpku4aqm2etwVDaw5pfYEYj238xSWAddGBvJWVehL%2Bq3LbLsFmyE3RAfNbta0gx3uvhV3mh9WJwiAROswnNPw%2F4Ug7NlXkV7gwHfMmXu62%2BP8%2BbCs7ZTzDlnDnPfkYDVWl0Sw4avQOr5O8PDhtMdPKQHG23RbFlW7rVHEkbhBQJt%2FKjMpT%2BrJbTtjIIayI6u5TdX0kxGtZ3Xw8mGc9WLuvb3u9tnszG57X7Lv9AoKjk5B%2Fql02q0ZAFUZ%2F92qOjZykAyEK53179tIe7476EZGhYjRVHunwDII19iNa%2FqJMLXHN7VRccuLF7kFtzQskKkaZErd%2FN1SgasSCwJ%2Bt4FLONGqMg86l6WgAtOFpXxJxjJqTdVCo9NglH0JGN7v843S5QsrU%2FxuT4nMWG%2FqPIvWu2sLmPv5fE0UmWc2c7hzs13T2Q8Yu%2Bv3PcZ%2FkmPzhF8Oyjm0vh%2BJ3J%2FiTiHf2QNicTSONzenqswnwDP60oo5qXLE1DPySufQQ9%2BGJLHfsSPj46wd7yPQekc5tz0kKw4B75NP9GbMI6CDMccsYAheZCZrIWPKaazWaipg%2FLPL4DFLI9UbqCoLtfAeFZEpUxU8QTepmkMaak1bD1wnM53LivQoxmuYcVjIWOusRhPPlkceGYD1oSAhY3SAbUCT1SX9DkY0IZiiN5Nr3ZRSqao1yxWk6gwUSmrN85pPikf4vtaLXCR%2Fs1NAfnJbBwlAcAw8uyK31fhHETFRgxcnaYjVOEuSrhq32ezgvpnxlfwfozyZK4PHlEwXgbXcduU3c%2Fo6%2FjJmbCixOO9Th21mYsoZEj8p%2BD568bENq03nSY86Cpjd%2BmbHQtZXFXOD4yLafNTskgy12EZQdUxnG7vCxKrSUbmvwntiFzSxr6xhXw%2FyOLvzkw403ny3gfXFFWAlVIdK8uR5%2Bev8a3M5zuCSC9Y2mHPTb&maxtime=%222024-03-11%2021:07:47%22
```
[the remark this answers is not in the document]
Comment (LudovicMalot) [reply src/index.ts:13]: very soon to be expired so i'd rather leave it here as an accurate exemple.

File: src/utils/wishImport.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
+import axios from "axios";
```
[the remark this answers is not in the document]
Comment (LudovicMalot) [reply src/utils/wishImport.ts:1]: Axios will be more flexible and easier for all the data fetching. We could use fetch but I feel like it's a bit more maintainable and readable with Axios

File: src/utils/wishImport.ts (hunk 2 above)
Comment (LudovicMalot) [reply src/utils/wishImport.ts:66]: five seems to be a soft spot to be sure not to get banned.i've added a random delay

**Review by maalni — APPROVED** [review]

**Pull request merged**